rest-client/client-common/src/main/java/io/apicurio/registry/rest/client/impl/RegistryClientImpl.java

```diff
@@ -427,6 +427,24 @@ private String normalizeGid(String groupId) {
         return groupId == null ? "default" : groupId;
     }
 
+    private String encodeToBase64(String toEncode) {
+        return Base64.getEncoder().encodeToString(toEncode.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private Map<String, String> headersFrom(String version, String artifactName, String artifactDescription) {
+        final Map<String, String> headers = new HashMap<>();
```

Comment:
Good. Why not adding the other headers to this method as well?

Reply:
That is not a bad idea. I was wondering about it before but I have decided this way because these headers are only ones that would repeat more times. ArtifactID and ArtifactType is createArtifact specific and cannot be used anywhere else.
But I would agree with adding them in this method if you think it is better approach.

Reply:
@carlesarnal

Reply:
Hm, you're right, no, let's leave this as is.